dpgen2/exploration/deviation/deviation_spin.py

```
@@ -0,0 +1,127 @@
+            self.MAX_DEVI_MF,
+            self.MIN_DEVI_MF,
+            self.AVG_DEVI_MF,
+        ), f"Error: unknown deviation name {name}"
+
+    def _add(self, name: str, deviation: np.ndarray) -> None:
+        assert isinstance(
+            deviation, np.ndarray
+        ), f"Error: deviation(type: {type(deviation)}) is not a np.ndarray"
+        assert len(deviation.shape) == 1, (
+            f"Error: deviation(shape: {deviation.shape}) is not a "
+            + f"one-dimensional array"
```

Ruff

f-string without any placeholders

Remove extraneous `f` prefix

(F541)